trio/_core/_run.py

```diff
@@ -333,30 +372,30 @@ class NurseryManager:
 
     @enable_ki_protection
     async def __aenter__(self):
-        assert currently_ki_protected()
-        self._scope_manager = open_cancel_scope()
-        scope = self._scope_manager.__enter__()
-        self._nursery = Nursery(current_task(), scope)
+        self._scope = CancelScope._create(deadline=inf, shield=False)
+        self._nursery = Nursery(current_task(), self._scope)
         return self._nursery
 
     @enable_ki_protection
     async def __aexit__(self, etype, exc, tb):
-        assert currently_ki_protected()
-        try:
-            await self._nursery._nested_child_finished(exc)
-        except BaseException as new_exc:
-            try:
-                if self._scope_manager.__exit__(
-                    type(new_exc), new_exc, new_exc.__traceback__
-                ):
-                    return True
-            except BaseException as scope_manager_exc:
-                if scope_manager_exc == exc:
-                    return False
-                raise  # scope_manager_exc
-            raise  # new_exc
+        new_exc = await self._nursery._nested_child_finished(exc)
+        if new_exc:
+            scope_exc = self._scope._close(new_exc)
```

Comment:
You should move this up a line, out of the `if new_exc`. The `_close` method needs to be called unconditionally, and it already accepts `None` as an argument.

Reply:
Oops, thank you!  Actually the whole `if` block shouldn't be there.